generators/sram-bist/src/test/scala/BistTopSpec.scala

```diff
@@ -738,7 +736,183 @@ class BistTopSpec extends AnyFlatSpec with ChiselScalatestTester {
       testhelpers.c.io.bistSignature.expect(misrModelB.state.U)
     }
   }
+  it should "work with 16 data bit chiseltest SRAMs" in {
+    test(
+      new BistTop(
+        new BistTopParams(
+          Seq(new SramParams(16, 8, 64, 16)),
+          new ProgrammableBistParams(
+            patternTableLength = 4,
+            elementTableLength = 4,
+            operationsPerElement = 4
+          )
+        )
+      )(
+        new WithChiseltestSrams(ChiseltestSramFailureMode.none)
+      )
+    ).withAnnotations(Seq(WriteVcdAnnotation)) { c =>
+
+      // TODO: Test on SRAMs with smaller than 32 data width to make sure we aren't checking too many bits.
+      val testhelpers = new BistTopTestHelpers(c)
+      c.clock.setTimeout(
+        (testhelpers.maxCols + 1) * (testhelpers.maxRows + 1) * 4 * 4 * 3
+      )
+      val testSramMethod = (executeFn: () => Unit) => 
+        // Test write to small SRAM.
+        testhelpers.populateSramRegisters(
+          0.U,
+          "hab6f6bbd".U,
+          "hf".U,
+          true.B,
+          0.U,
+          SramSrc.mmio,
+          SaeSrc.int
+        )
+        executeFn()
+
+        testhelpers.populateSramRegisters(
+          0.U,
+          0.U,
+          0.U,
+          false.B,
+          0.U,
+          SramSrc.mmio,
+          SaeSrc.int
+        )
+        executeFn()
+        testhelpers.c.io.dout.expect("h00006bbd".U)
```

Comment:
We don't actually mask the top level IO dout, so I think it would be more correct to only check the second half. Probably doesn't matter too much, but I don't think `dout` should be masked since it can be masked at a higher level.